pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -244,3 +249,20 @@ func (c *Reconciler) getServiceAccount(pr *v1alpha1.PipelineRun) (string, error)
 	}
 	return pp.Spec.ServiceAccount, nil
 }
+
+func (c *Reconciler) reconcilePipelineRunStatus(pr *v1alpha1.PipelineRun, state []*resources.PipelineRunTaskRun) {
+	// reconcile pipeline run status first to see if we need to run any more tasks.
+	prStatus := resources.GetPipelineConditionStatus(pr.Name, state, c.Logger)
+	pr.Status.SetCondition(prStatus)
+	c.Logger.Infof("PipelineRun %s status is being set to %s", pr.Name, pr.Status)
+}
+
+func (c *Reconciler) isPipelineRunComplete(pr *v1alpha1.PipelineRun) bool {
+	// Check if Pipeline Run is already marked as completed or failed
+	if s := pr.Status.GetCondition(duckv1alpha1.ConditionSucceeded); s != nil && s.Status != corev1.ConditionUnknown {
```

Comment:
nit: ducktype provides `IsUnknown()` function to check `s.Status != corev1.ConditionUnknown`. It might be good to use ducktype functions.

Reply:
oh nice, i had no idea those existed!

Reply:
Thanks! will add this.